File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/SpaceCityToUI.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author Profesor
 */
public class SpaceCityToUI extends SpaceStationToUI {

  // MIGUEL: He añadido este atributo para que el ToUI sea completo
  // Modificado el constructor y añadido un consultor
    
  ArrayList<SpaceStationToUI> collaborators = new ArrayList();
  
  SpaceCityToUI (SpaceCity city) {
    super (city);
    for (SpaceStation s : city.getCollaborators()) {
        collaborators.add (new SpaceStationToUI(s));
    }
  }
  
  @Override
  public String getName () {
    return super.getName()+" (CIUDAD ESPACIAL)";
  }
  
  public ArrayList<SpaceStationToUI> getCollaborators() {
      return collaborators;
  }
}

File: C2/PDOO/Practicas/DeepSpaceRuby/lib/SpaceCityToUI.rb

require_relative 'SpaceStationToUI'

module Deepspace

# 24.5.17 - Translation from Java
# @author Profesor
  
class SpaceCityToUI < SpaceStationToUI 
  
  attr_reader :collaborators
  
  def initialize (city) 
    super(city)
    @collaborators=Array.new
    for c in city.collaborators do 
      @collaborators << SpaceStationToUI.new(c)
    end
  end
  
  #Override
  def name () 
    return super + " (CIUDAD ESPACIAL)"
  end
  
  #Override
  def to_s
    out = super
    out += "\n ------- My Collaborators are:"
    for c in @collaborators do
      out += "\n --- Collaborator --- \n"
      out += c.to_s
    end
    out += "\n ------- No More Collaborators \n"
    return out
  end

end # class

end # module
